runtime/etc/mli/mli_client_runtime.c

```diff
@@ -119,10 +119,13 @@ void chpl_library_finalize(void) {
   if (finalized) { return; }
   finalized = 1;
 
-  // TODO: Shut down the server and handle errors.
-  if (0) {
-    int64_t shutdown = -1;
-    chpl_mli_push(chpl_client.main, &shutdown, sizeof(shutdown), 0);     
+  {
+    int64_t shutdown = CHPL_MLI_ERROR_SHUTDOWN;
```

Comment:
Your client clean shutdown number and error shutdown number appear to be the same, how do we distinguish between a clean shutdown and a shutdown due to an error?

Reply:
So perhaps that's a failing in my terminology. All the "directives" are called error codes but not all of them are "errors", so to speak.

Currently, when the client wants to shut down, it sends the shutdown "error code", and then the server responds with the  shutdown error code in turn. I suppose the server could respond with "none" as well. What matters currently is the ACK, not the value of it (thus the comment).

Reply:
I can rename the "error" codes to reflect different semantics here now, if you'd like.